sdk/core/core-http/src/policies/deserializationPolicy.ts

```diff
@@ -353,15 +353,28 @@ function parse(
     const contentComponents: string[] = !contentType
       ? []
       : contentType.split(";").map((component) => component.toLowerCase());
-    if (
-      contentComponents.length === 0 ||
-      contentComponents.some((component) => jsonContentTypes.indexOf(component) !== -1)
-    ) {
+    let isJSONResponse = contentComponents.some(
+      (component) => jsonContentTypes.indexOf(component) !== -1
+    );
+    let isXMLResponse = contentComponents.some(
+      (component) => xmlContentTypes.indexOf(component) !== -1
+    );
+    // Attempt to parse the response body as JSON when the content-type doesn't match any known JSON/XML types
+    if (contentComponents.length === 0 || (!isJSONResponse && !isXMLResponse)) {
+      try {
+        JSON.parse(text);
+        isJSONResponse = true;
+      } catch (error) {
+        // If fail to parse as JSON, assume the response body is XML
```

Comment:
why would we assume XML? What if it's something like JSONC or NDJSON?

Reply:
To match the original behavior, we can again check for XML. So when there is no content-type, we 
- If is JSON, parse as JSON. Used to parse as JSON.
- If XML, parse as XML. Used to throw while attempting to parse it as JSON.
- If else, returned the raw response body. Used to return the raw response body.